mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -0,0 +1,633 @@
+        daily process.  It will generate the echograms based on those inputs
+        and upload the echograms to the raw data server.
+
+        :return:
+        """
+
+        # Create the temporary data file directory.
+        self.temp_directory = join(expanduser(USER_HOME), TEMP_DIR)
+        if not exists(self.temp_directory):
+            mkdir(self.temp_directory)
+
+        # Create the echograms for the zplsc instruments of each subsite.
+        for subsite in self.subsites:
+            zplsc_24_subsite_prefix = subsite + '-'
+
+            try:
+                deployments = self.get_deployment_dirs(subsite)
+            except urllib2.HTTPError:
+                continue
+
+            for deployment in deployments:
+                zplsc_24_deployment_prefix = zplsc_24_subsite_prefix + 'R' + str(deployment) + '-'
+
+                try:
+                    echogram_dates, date_dirs_url = self.get_date_dirs(subsite, deployment)
+                except urllib2.HTTPError:
+                    continue
+
+                for date_dir, entire_month in echogram_dates.items():
+                    self.zplsc_24_datafile_prefix = zplsc_24_deployment_prefix + 'sn' + self.serial_num + '-'
+
+                    if entire_month:
+                        number_of_days_in_the_month = calendar.monthrange(date_dir.year, date_dir.month)[1]
+                        for day in range(number_of_days_in_the_month):
+                            echogram_date = date_dir + timedelta(days=day)
+
+                            # Aggregate the 24 raw data files for the given instrument to 1 24-hour data file.
+                            try:
+                                zplsc_24_datafile, zplsc_echogram_file = self.aggregate_raw_data(date_dirs_url,
+                                                                                                 echogram_date)
+                                if not zplsc_24_datafile:
+                                    continue
+
+                            except urllib2.HTTPError:
+                                continue
+
+                            # Send the 24-hour raw data file to the zplsc C Series parser to generate the echogram.
+                            with open(zplsc_24_datafile) as file_handle:
+                                parser = ZplscCParser(CONFIG, file_handle, self.rec_exception_callback)
+                                parser.create_echogram()
+
+                            self.store_echogram(zplsc_echogram_file)
+                    else:
+                        # Aggregate the 24 raw data files for the given instrument to 1 24-hour data file.
+                        try:
+                            zplsc_24_datafile, zplsc_echogram_file = self.aggregate_raw_data(date_dirs_url, date_dir)
+
+                            if not zplsc_24_datafile:
+                                continue
+
+                        except urllib2.HTTPError:
```

Comment:
debug out?

Reply:
Removed the try catch block around aggregate_raw_data to catch HTTPError, because it will not throw that exception.  I added a warning of the "if not zplsc_24_datafile" if it was unable to aggregate the 24 1-hour files.